src/qibolab/identifier.py

```diff
@@ -0,0 +1,63 @@
+from enum import Enum
+from typing import Annotated, Optional, Union
+
+from pydantic import Field, TypeAdapter, model_serializer, model_validator
+
+from .serialize import Model
+
+QubitId = Annotated[Union[int, str], Field(union_mode="left_to_right")]
+"""Type for qubit names."""
+
+
+# TODO: replace with StrEnum, once py3.10 will be abandoned
+# at which point, it will also be possible to replace values with auto()
+class ChannelType(str, Enum):
+    """Names of channels that belong to a qubit.
+
+    Not all channels are required to operate a qubit.
+    """
+
+    PROBE = "probe"
+    ACQUISITION = "acquisition"
+    DRIVE = "drive"
+    DRIVE12 = "drive12"
+    DRIVE_CROSS = "drive_cross"
+    FLUX = "flux"
```

Comment:
As explained above, it is, exactly because we need to navigate the `Qubit` object to access the channel.

You can consider the `ChannelId` `qubit/chtype` to be the path:
```py
getattr(platform.qubits[qubit], chtype)
```